### DISCUSSION TITLE
How can I start multiple selections (cursors) in a text editor?

### DISCUSSION THREAD

**Discussion by lgrammel** [microsoft/vscode-discussions#39] [Extension Development QnA] [answered]
I want to select several text ranges and start multi-cursor editing throught the API. I looked into the docs and `selections` on `TextEditor` is `readonly`. Is there recommended way to start multi-cursor editing?

**Comment by usernamehw** [comment, marked as answer]
You set multiple cursors by assigning `selections`

```ts
textEditor.selections = [
	new Selection(0, 1, 0, 1),
	new Selection(0, 2, 0, 2),
];
```

---

TypeScript types show:

```ts
selections: readonly Selection[];
```

It would have been `readonly` property if it was a prefix:

```ts
readonly selections: readonly Selection[];
```

**Reply by lgrammel** [reply]
Thanks, I misread that!
